jina/executors/evaluators/rank/recall.py

```diff
@@ -18,14 +18,13 @@ def evaluate(self, actual: Sequence[Any], desired: Sequence[Any], *args, **kwarg
         :param desired: the expected documents matches ids sorted as they are expected
         :return the evaluation metric value for the request document
         """
+        if not desired:
+            """TODO: Agree on a behavior"""
+            return 0.0
+
         ret = 0.0
-        for doc_id in desired[:self.eval_at]:
-            if doc_id in actual:
+        for doc_id in actual[:self.eval_at]:
+            if doc_id in desired:
```

Comment:
Just to make sure, we agree then that all the matches here are considered as `relevant` right?

Reply:
I think it should be documented expicitly that all the documents in `desired` will be considered as `relevant`

Reply:
That is my understanding of `desired`, yes.

Reply:
Then rename it to `relevant_docs`?

Reply:
it is okey as it is I think, and we have `desired` all across the evaluators so it would be okey.